Swagger spec file for the dataset api

Matt-Guest opened the pull request:
Includes endpoints to query datasets and dimensions. Also contains models of the json messages to be returned.

This is the initial work towards the api, this will change over
time.

CarlHembrough commented on swagger.yaml:
```diff
@@ -0,0 +1,262 @@
+      produces:
+      - "application/json"
+      responses:
+        200:
```
consider adding other response codes for various error responses.

CarlHembrough commented on swagger.yaml:
```diff
@@ -0,0 +1,262 @@
+        200:
+          description: "A json list contain datasets which have been published"
+          schema:
+            $ref: '#/definitions/ArrayOfDatasets'
```
I imagine this list will be paginated. Consider adding details of pagination to the spec. Not sure if this is something swagger supports?

Matt-Guest commented:
I updated the swagger spec to include additional error codes 400, 404 and 500. Added pagination and a search endpoint. Can you please review the changes.
